### PR TITLE
Maskmaking documentation

### PR DESCRIPTION
**Pull request by Lvulis**
- Documentation for (SAM)
- Included sentence about using individual monthly water masks from GSW

Several commits are just me pulling from upstream.

### CODE REVIEW

**Comment by elbeejay** [comment]
Let me know if/when this is done and I can read over the changes once before merging.

<br>

> Several commits are just me pulling from upstream.

Just a note (not that we're picky for this project), but general best practice is to create a separate "feature branch" in your fork that you'd open a pull request from. Avoiding commits directly to your fork's "master" helps minimize merge commits as it lets you keep your "master" branch cleaner and just something that tracks the canonical "VeinsOfTheEarth/master" branch. Changes via commits to feature branches stick out from there and "merge commits" typically happen when those changes are incorporated into the "VeinsOfTheEarth/master" via PR.

**Comment by Lvulis** [comment]
> Let me know if/when this is done and I can read over the changes once before merging.

This is done now! Just caught an extra space when reviewing the changes in GH.

> Just a note (not that we're picky for this project), but general best practice is to create a separate "feature branch" in your fork that you'd open a pull request from. Avoiding commits directly to your fork's "master" helps minimize merge commits as it lets you keep your "master" branch cleaner and just something that tracks the canonical "VeinsOfTheEarth/master" branch. Changes via commits to feature branches stick out from there and "merge commits" typically happen when those changes are incorporated into the "VeinsOfTheEarth/master" via PR.

Got it, thanks for the advice. Was wondering how that's handled normally.

File: docs/source/maskmaking/index.rst [hunk 1]
```diff
@@ -106,7 +106,7 @@ Here are some resources that either provide masks or tools for you to make your
...
-  - `Global Surface Water Dataset <https://global-surface-water.appspot.com/>`_ - provides all water pixels in the Landsat archive as monthly global images and as integrated-through-time images. For example, can threshold on the "Occurrence" product to make a mask. Use `Google Earth Engine <https://developers.google.com/earth-engine/datasets/catalog/JRC_GSW1_2_GlobalSurfaceWater>`_ to access and create your masks.
+  - `Global Surface Water Dataset <https://global-surface-water.appspot.com/>`_ - provides all water pixels in the Landsat archive as monthly global images and as integrated-through-time images. For example, can threshold on the "Occurrence" product to make a mask. Use `Google Earth Engine <https://developers.google.com/earth-engine/datasets/catalog/JRC_GSW1_4_GlobalSurfaceWater>`_ to access and create your masks. You can also access individual monthly water masks `here <https://developers.google.com/earth-engine/datasets/catalog/JRC_GSW1_4_MonthlyHistory>`.
```
Comment (elbeejay) [inline docs/source/maskmaking/index.rst:109]: Is "you" or "one" missing after "For example,"? 

"For example, **you** can threshold..." or "For example, **one** can threshold..."

File: docs/source/maskmaking/index.rst [hunk 2]
```diff
@@ -120,6 +120,7 @@ Here are some resources that either provide masks or tools for you to make your
 
 - `DeepWaterMap  <https://github.com/isikdogan/deepwatermap>`_ is a trained deep convolutional neural network that you can apply to Landsat/Sentinel multispectral imagery to create your own masks. You can also improve DeepWaterMap's base model by adding more training data. Requires some knowledge of Tensorflow.
 
+- A recent development as of April 2023 is the release of the *Segment Anything Model* `(SAM) <https://segment-anything.com/>`_ from Meta Research which allows rapid segmentation of any image, including satellite imagery. The SAM model has an API in Python via `segment-geospatial <https://github.com/opengeos/segment-geospatial>`_. *segment-geospatial* and could be used for rapid identification of water from various satellite imagery sources, e.g. Planet or Landsat.
```
Comment (elbeejay) [inline docs/source/maskmaking/index.rst:123]: Is there a superfluous "and" after *segment-geospatial*?

"*segment-geospatial* and could..."

**Review by elbeejay — CHANGES_REQUESTED** [review]
2 minor questions / suggestions

File: docs/source/maskmaking/index.rst (hunk 2 above)
Comment (elbeejay) [reply docs/source/maskmaking/index.rst]: I usually build the documentation locally. Just checked now and it doesn't look like we actually document how to do that 😬. The shell commands from the repository root are:

```
cd docs
make docs
make html
```

which is what we do for the actual automated deployment ([L39](https://github.com/VeinsOfTheEarth/RivGraph/blob/38f2a252a8f64ec88368d162b9794e1e5ff02b4c/.github/workflows/docs.yml#L39)), but we should definitely document the process for installing the required libraries and doing that local build.

**Review by elbeejay — APPROVED** [review]
Looks great

**Pull request merged**